feat: Первая версия проекта

Mity1440 opened the pull request.

Mity1440 committed:
feat: Первая версия проекта помещена под версионный контроль

Mity1440 committed:
chore: Добавлена лицензия

petrelevich commented on src/main/java/com/dmakarevich/yellow_collector/controllers/GetInfoRestController.java:
```diff
@@ -0,0 +1,31 @@
+        this.processor = processor;
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/getInfo")
```
Лучше использовать более специфичную аннотацию PostMapping

petrelevich commented on src/main/java/com/dmakarevich/yellow_collector/controllers/PushReportRestController.java:
```diff
@@ -0,0 +1,47 @@
+
+    private final PushReportProcessor processor;
+
+    @Autowired
```
на конструкторе эат аннотация не нужна

petrelevich commented on src/main/java/com/dmakarevich/yellow_collector/controllers/PushReportRestController.java:
```diff
@@ -0,0 +1,47 @@
+        try {
+            processor.processReportFile(getMultipartFileFromRequest(request));
+        } catch (Exception e) {
+            e.printStackTrace();
```
стоит использовать логирование.

petrelevich commented on src/main/java/com/dmakarevich/yellow_collector/db/model/main/ServerInfo.java:
```diff
@@ -0,0 +1,27 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class ServerInfo {
```
очень большое кол-во моделей, разбросанных по всему коду, вызывает подозрения в переусложнении и дублировании.
Например, вот эти классы точно должны быть разными:
public class FileServerInfo
public class ServerInfo
?
По полями они полностью совпадают и названия похожи.

petrelevich commented on src/main/java/com/dmakarevich/yellow_collector/db/services/ReportProcessorDBServiceImpl.java:
```diff
@@ -0,0 +1,71 @@
+    private final ErrorInfoReportRepository errorInfoReportRepository;
+    private final ErrorInfoDetailRepository errorInfoDetailRepository;
+    private final ErrorInfoStackDetailRepository errorInfoStackDetailRepository;
+    private final ReportInfoConverter converter;
```
потерялся дженерик.

petrelevich commented on src/main/java/com/dmakarevich/yellow_collector/services/report/runners/ReportProcessorRunnerImpl.java:
```diff
@@ -0,0 +1,92 @@
+
+        this.workerPool = Executors.newFixedThreadPool(this.runnersProperties.getWorkerPoolSize());
+        this.filesProcessingQueue
+                = new ArrayBlockingQueue(this.runnersProperties.getMaximumFileProcessingQueueSizePerIteration());
```
нехватает джененрика.

petrelevich commented on src/main/java/com/dmakarevich/yellow_collector/services/report/runners/ReportProcessorRunnerImpl.java:
```diff
@@ -0,0 +1,92 @@
+    private void fillFilesProcessingQueue() throws IOException {
+
+        var reportFiles = finder.find(runnersProperties.getMaximumFileProcessingQueueSizePerIteration());
+        reportFiles.stream().forEach(filesProcessingQueue::offer);
```
что будет, если offer не сможет добавить файл в очередь?

Mity1440 committed:
refactor: Переработан проект

Основные изменения
- Многомодульный gradle проект
- В качестве БД используется elasticsearch, для задач поиска подходит гораздо лучше чем MongoDB

Mity1440 committed:
feat: Добавлен модуль general-requester, отвечающий за обработку клиентских запросов и представляющий уупрощенное api данным отчетов об ошибках

Mity1440 committed:
feat: Сформирован минимально рабочий вариант

Mity1440 commented:
С момента первого прототипа - изменил взгляд на реализацию проекта в целом.
